pkg/ebpf/perf.go

```diff
@@ -15,42 +15,32 @@ import (
 	manager "github.com/DataDog/ebpf-manager"
 )
 
-// PerfHandler wraps an eBPF perf buffer
-type PerfHandler struct {
-	DataChannel  chan *DataEvent
-	LostChannel  chan uint64
-	RecordGetter func() *perf.Record
-	once         sync.Once
-	closed       bool
+var recordPool = sync.Pool{
+	New: func() interface{} {
+		return new(perf.Record)
+	},
 }
 
-// DataEvent is a single event read from a perf buffer
-type DataEvent struct {
-	CPU  int
-	Data []byte
-
-	r *perf.Record
-}
+var _ EventHandler = new(PerfHandler)
```

Comment:
I didn't know this pattern before. I suggest adding a comment to briefly explain what this does, since not every reader knows of it.

Reply:
Good point. Done!